option.go

```diff
@@ -22,6 +22,18 @@ func Routes(rr []http.Route) OptionFunc {
 	}
 }
 
+// Middlewares option for adding generic middlewares to the default HTTP component.
+func Middlewares(mm []http.MiddlewareFunc) OptionFunc {
+	return func(s *Service) error {
+		if len(mm) == 0 {
+			return nil
```

Comment:
if the user supplies middlewares via the option he should provide at least one. this should be a actual error.